File: csharp/byps/BSerializer_14.cs
/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace byps
{
    public class BSerializer_14 : BSerializer_12
    {
	    public readonly static new BSerializer instance = new BSerializer_14();

	    public BSerializer_14() 
            :base(BRegistry.TYPEID_SET)
        {
	    }

	    public BSerializer_14(int typeId) 
            :base(typeId)
        {
	    }

        public override Object read(Object obj1, BInput bin1, long version)
        {
            BInputBin bin = ((BInputBin)bin1);
            BBufferBin bbuf = bin.bbuf;
            int n = bbuf.getLength();
            HashSet<Object> arr = new HashSet<Object>();
            for (int i = 0; i < n; i++)
            {
                Object obj = (Object)bin.readObj(false, null);
                arr.Add(obj);
            }
            return arr;
        }

        public override void write(Object obj1, BOutput bout1, long version)
        {
            BOutputBin bout = ((BOutputBin)bout1);
            BBufferBin bbuf = bout.bbuf;
            HashSet<Object> arr = (HashSet<Object>)obj1;
            int n = arr.Count;
            bout.bbuf.putLength(n);
            foreach (Object obj in arr)
            {
                bout.writeObj(obj, false, null);
            }
        }


    }
}

File: java/byps/src-ser-bin/byps/BSerializer_14.java
package byps;
/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */
import java.util.Collection;
import java.util.HashSet;

import byps.BException;
import byps.BInput;
import byps.BRegistry;
import byps.BSerializer;

public class BSerializer_14 extends BSerializer_12 {

	public final static BSerializer instance = new BSerializer_14();

	public BSerializer_14() {
		super(BRegistry.TYPEID_SET);
	}

	public BSerializer_14(int typeId) {
		super(typeId);
	}

	@Override
	public Collection<Object> internalCreate(final BInput bin1) throws BException {
		return new HashSet<Object>();
	}
	
	
}